tg_bot/modules/sql/locks_sql.py

```diff
@@ -216,6 +231,64 @@ def get_restr(chat_id):
         SESSION.close()
 
 
+def get_whitelist(chat_id):
+    return CHAT_WHITELIST.get(str(chat_id), {})
+
+
+def add_whitelist(chat_id, url):
+    global CHAT_WHITELIST
+    with WHITELIST_LOCK:
+        url = re.search(r'(^http:\/\/|^https:\/\/|^ftp:\/\/|^)(www\.)?(\S*)', url, flags=re.I).group(3).lower()
+        if url.endswith('/'):
+            url = url[:-1]
+        prev = SESSION.query(URLWhitelist).get((str(chat_id), url))
+        if not prev:
+            whitelisted = URLWhitelist(str(chat_id), url)
+            SESSION.add(whitelisted)
+            SESSION.commit()
+        chat_whitelist = CHAT_WHITELIST.setdefault(str(chat_id), {})
```

Comment:
shouldnt this be indented too? since `if not prev`, this is already loaded?

Reply:
This is more a case of "double bagging" :-)  If for whatever reason the URL was in the DB but not in the dictionary, now it **sure** is.

At least the `return True` statement I would leave unindented, so the bot's confirmation message will include the URL even if it was already previously added.